src/classes/GE_FormRendererService.cls

```diff
@@ -134,7 +136,8 @@ public with sharing class GE_FormRendererService {
                         // Update the element's required by doing an OR between the element's required, DI source field, target field
                         // Update the element's required to make sure we are respecting the source/target/template builder's required
                         // with the most up to date information. 
-                        element.required = element.required || fieldMapping.Source_Field_Required;
+                        element.required = element.required || 
+                            (fieldMapping.Source_Field_Required && fieldMapping.Source_Field_Data_Type != BOOLEAN_DATA_TYPE);
```

Comment:
Very minor, but would be nice for this to fit on one line in the github review window